### PR TITLE
Fix flex.rb source_url and update sha1 sum of new target file

### PR DESCRIPTION
**Pull request by Brunux**
Fix the target download file for flex package refer to issue #179

### CODE REVIEW

File: packages/e2fsprogs.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    system "CFLAGS=-I/tools/include"
+    system "PKG_CONFIG_PATH=/usr/local/bin/pkg-config"
+    system "../configure --prefix=/usr/local --bindir=/usr/local/bin --enable-elf-shlibs --disable-uuidd --disable-fsck"
+    system "make"
+  end
+
+  def self.install
+    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install-libs"
+    system "ls /usr/local/sbin/ | sudo xargs -I {} ln -sv /usr/local/sbin/{} /usr/local/bin/{}"
```
Comment (Brunux) [inline packages/e2fsprogs.rb]: Does 'sudo' is the right way to do it?

**Comment by lyxell** [comment]
I don't really like the idea of encouraging users to mount their partitions with rw-flags since this breaks the security model of the OS. A better solution would be to add path changes to ~/.bash_profile if that is possible.

Also, regarding the comment about wierd behaviour in vim, compatible-mode is actually the **default** behaviour, I feel like the comment is a bit misleading.

Another thing, wouldn't symlinking /usr/local/sbin to /usr/local/bin break package uninstallation, leaving broken symlinks?

Lastly, I'm a bit concerned with whether it is a good idea to introduce compiled files contributed by "normal users" without the ability to review them.

**Comment by Brunux** [comment]
Hi @lyxell about mount rw-flags I agree with you I believe add the necessary to $PATH is a better approach using ~/.bash_profile is sourced by bash when started in interactive login mode, so it should work since currently there are no X applications (btw I'm working to bring X).

About behaviour in vim you are right that is the default behaviour (which I don't find useful) the idea was to give a message to the user in my experience most people use vim with batteries meaning they have full working (pre-configure) vim. So maybe just give a warning the users so they know ~/.vimrc needs to be configured.

About symlinks for precompiled package (which include symliks) current the create_package.sh only finds files and dirs, it doesn't  find for symliks (find . -type l) so yes it leaves symliks hunging around after removing a package. Adding the necessary paths to PATH fix this symlinking issue.

So basically what I purpose is:

1. Give a message to the user about vim to let him/she know about ~/.vimrc needs proper setup.
2. Add /usr/local/sbin to $PATH in some way

Finally I totally agree with this

> Lastly, I'm a bit concerned with whether it is a good idea to introduce compiled files contributed by "normal users" without the ability to review them.

So how do you purpose to review them?
Cheers

**Comment by lyxell** [comment]
> So how do you purpose to review them?

I'm not sure, there's probably a few ways, but a pretty intricate infrastructure would probably be required to automate it, so in my opinion installing from source in the recipes is better for now.

**Comment by Brunux** [comment]
Some times the official source_url is 404 so it would be a good idea to have those backed-up. You can always check-sum the backed-up files against the 'official source' one even check the code itself if you feel so.

Been honest here I'm not trying to hack nobody (is my name on the line), I needed those package for my own that's why I build those and I though why not share those since I'm using some binaries from other packages.

 So this brings this question. Are you open to accept binaries? If so when? and how? 

meanwhile I will remove binaries from formulas (even if I believe is a waste of time to recompile those) until you guys trust my binaries

**Comment by lyxell** [comment]
I'm in no way an official spokesman for this project, I'm just putting in my two cents to open an discussion. I trust your intentions, and there's a few binaries in the project already, so it might not be an issue at all. However, it would be interesting to know the opinions of the community and the contributors regarding this issue.

**Comment by Brunux** [comment]
Thank you for the feedback it's well appreciated

**Comment by thedamian** [comment]
I think you've had too much vino @brunux and checked in changes you didn't meant to check in.

Try getting to at least the #2 spot before you submit a change in the name of this projects name.

Lay off the drugs my friend! No Bueno!

**Comment by thedamian** [comment]
Oooihhhh you're trying to get @skycocker to see your auggestions. Got it.

I see what you did there..... Very sneaky. Ok.

I'll watch from here and see if it worked...

**Comment by Brunux** [comment]
Sorry I forgot there was this PR ongoing I meant those changes for my own repo I believe this is moving to slow for me (hope you agree with this too). So I'm going to move it in a different direction.
btw I don't like wine but I love beer "Corona" specially.
